### PR TITLE
EOS:17953 - Update s3 mini-provisioning scripts as per updated confstore keys.

### PR DESCRIPTION
**Pull request by amitwac2608**
Signed-off-by: Amit Kumar <amit.kumar@seagate.com>

### CODE REVIEW

File: scripts/provisioning/s3.config.conf.tmpl [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (nileshgovande) [inline scripts/provisioning/s3.config.conf.tmpl:1]: Copyright

File: scripts/provisioning/s3.init.conf.tmpl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (nileshgovande) [inline scripts/provisioning/s3.init.conf.tmpl:1]: Copyright

File: scripts/provisioning/setupcmd.py [hunk 3]
```diff
@@ -44,65 +43,68 @@ def __init__(self, config: str):
...
+        self.s3_confkeys_store.get_config(
+          'CONFSTORE_LDAPADMIN_USER_KEY'))
 
-      self.ldap_user = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_LDAPADMIN_USER_KEY'))
+      encrypted_ldapadmin_pass = self.provisioner_confstore.get_config(
+        self.s3_confkeys_store.get_config('CONFSTORE_LDAPADMIN_PASSWD_KEY'))
 
-      encrypted_rootdn_pass = self.s3confstore.get_config(localconfstore.get_config('CONFSTORE_ROOTDN_PASSWD_KEY'))
+      encrypted_rootdn_pass = self.provisioner_confstore.get_config(
+        self.s3_confkeys_store.get_config('CONFSTORE_ROOTDN_PASSWD_KEY'))
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:87]: @property
self.provisioner_confstore.get_config(self.s3_confkeys_store.get_config())

File: scripts/provisioning/s3.init.conf.tmpl (hunk 2 above)
Comment (amitwac2608) [reply scripts/provisioning/s3.init.conf.tmpl:1]: I realized it later that we cannot add copyright to this file as this file is a json file, (confstore would load it as json file) and we cannot add comments to json file.

File: scripts/provisioning/s3.config.conf.tmpl (hunk 1 above)
Comment (amitwac2608) [reply scripts/provisioning/s3.config.conf.tmpl:1]: I realized it later that we cannot add copyright to this file as this file is a json file, (confstore would load it as json file) and we cannot add comments to json file.

File: scripts/provisioning/setupcmd.py (hunk 3 above)
Comment (amitwac2608) [reply scripts/provisioning/setupcmd.py:91]: let me take this suggestion in next PR.

File: installhelper.sh [hunk 4]
```diff
@@ -95,6 +95,25 @@ cp -R scripts/haproxy/* $S3_INSTALL_LOCATION/install/haproxy
 
 # Copy the provisioning config
 cp scripts/provisioning/setup.yaml $S3_INSTALL_LOCATION/conf
+cp scripts/provisioning/s3.post_install.tmpl.1-node $S3_INSTALL_LOCATION/conf
+cp scripts/provisioning/s3.prepare.tmpl.1-node $S3_INSTALL_LOCATION/conf
```
Comment (nileshgovande) [inline installhelper.sh:99]: add empty sample files as well

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py [hunk 5]
```diff
@@ -42,11 +45,20 @@ def run():
...
-    numS3Instances = int(s3conf_store.get_s3instance_count(machine_id))
+    pvt_ip = provisioner_confstore.get_config(
+      local_confstore.get_config(
+        'CONFSTORE_PRIVATE_FQDN_KEY').format(machine_id))
+
+    pub_ip = provisioner_confstore.get_config(
+      local_confstore.get_config(
+        'CONFSTORE_PUBLIC_FQDN_KEY').format(machine_id))
+
+    numS3Instances = int(provisioner_confstore.get_config(
```
Comment (nileshgovande) [inline scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:58]: Try common property

File: scripts/provisioning/s3.config.tmpl.3-node [hunk 6]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
```
Comment (nileshgovande) [inline scripts/provisioning/s3.config.tmpl.3-node:1]: Has to be removed

**Review by vimalraghwani20 — APPROVED** [review]

File: installhelper.sh (hunk 4 above)
Comment (amitwac2608) [reply installhelper.sh:101]: Added

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py (hunk 5 above)
Comment (amitwac2608) [reply scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py]: @property decorator in python is used for getter-setter functions. I read it here: https://www.programiz.com/python-programming/property

But I have added new functions to encapsulate the above code.

File: scripts/provisioning/s3.config.tmpl.3-node (hunk 6 above)
Comment (amitwac2608) [reply scripts/provisioning/s3.config.tmpl.3-node]: done.

File: scripts/provisioning/s3.config.conf.tmpl (hunk 1 above)
Comment (amitwac2608) [reply scripts/provisioning/s3.config.conf.tmpl]: I have changed format from  json to yaml of the template files as suggested, and added the copyright header as well.

File: scripts/provisioning/s3.init.conf.tmpl (hunk 2 above)
Comment (amitwac2608) [reply scripts/provisioning/s3.init.conf.tmpl]: Added (after changing the format from json to yaml)

File: scripts/provisioning/setupcmd.py (hunk 3 above)
Comment (amitwac2608) [reply scripts/provisioning/setupcmd.py]: done

File: scripts/provisioning/setupcmd.py [hunk 7]
```diff
@@ -33,8 +33,7 @@ class SetupCmd(object):
   ldap_passwd = None
   rootdn_passwd = None
   cluster_id = None
-  server_nodes_count = 0
-  hosts_list = None
+  machine_id = None
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:36]: Need to write s3_provisioner_utils common class

Comment (amitwac2608) [reply scripts/provisioning/setupcmd.py:40]: taken care by @sachitanands in his PR.

**Review by nileshgovande — APPROVED** [review]

**Pull request merged**